src/Storages/MergeTree/MergeTreeWhereOptimizer.cpp

```diff
@@ -122,36 +122,61 @@ static bool isConditionGood(const ASTPtr & condition)
 
 void MergeTreeWhereOptimizer::analyzeImpl(Conditions & res, const ASTPtr & node, bool is_final) const
 {
-    if (const auto * func_and = node->as<ASTFunction>(); func_and && func_and->name == "and")
+    const auto * func = node->as<ASTFunction>();
+
+    if (func && func->name == "and")
     {
-        for (const auto & elem : func_and->arguments->children)
+        for (const auto & elem : func->arguments->children)
             analyzeImpl(res, elem, is_final);
     }
     else
     {
-        Condition cond;
-        cond.node = node;
+        const auto * func_tuple = func->arguments->children[0]->as<ASTFunction>();
```

Comment:
Why we considering first argument of some function, but not function itself? Also need to check if `func` is not null. And maybe add this branch to `if` above and leave `else` branch as is

Reply:
When you use `(k1, k2)=(v1,v2)`， the function is also `and`, so I have to make sure it's a tuple type filter by checking the argument of the function.

Reply:
I thought `k = v` and `(k1,k2)=(v1,v2)` are same function "equals"， so I have to check the arguments of function to make sure it's a tuple filter.  Because they're the same function , it means they will go to same branch if we use the  function name as the condition.